plutus-contract/src/Plutus/Contract/StateMachine.hs

```diff
@@ -331,10 +352,17 @@ runInitialise ::
     -- ^ The value locked by the contract at the beginning
     -> Contract w schema e state
 runInitialise StateMachineClient{scInstance} initialState initialValue = mapError (review _SMContractError) $ do
-    let StateMachineInstance{typedValidator, stateMachine} = scInstance
-        tx = mustPayToTheScript initialState (initialValue <> SM.threadTokenValue stateMachine)
-    let lookups = Constraints.typedValidatorLookups typedValidator
-    utx <- either (throwing _ConstraintResolutionError) pure (Constraints.mkTx lookups tx)
+    ownPK <- ownPubKey
+    utxo <- utxoAt (Ledger.pubKeyAddress ownPK)
```

Comment:
This works for now but in the future we should get rid of this as the `utxoAt` query generally returns too much data. Since we only need the bit that is specific to the `TxOutRef` of the thread token, maybe we could store this information in the thread token itself.

Reply:
I have added TODO comments